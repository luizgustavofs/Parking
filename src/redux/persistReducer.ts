import { Reducer } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';

const persistReducers = (reducers: Reducer<any>) => {
  const persistedReducer = persistReducer(
    {
      key: 'Parking',
      storage,
      whitelist: ['auth', 'session'],
    },
    reducers,
  );

  return persistedReducer;
};

export { persistReducers };
