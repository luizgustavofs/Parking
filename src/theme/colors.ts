export const colors = {
  default: '#000000',
  primary: '#4DD0E1',
  secondary: '#F2F2F2',
  tertiary: '#28DD91',
  quaternary: '#A769B2',
  error: '#FF0068',
  baseColor: '#FFFBE6',
  neutral: {
    _00: '#ffffff',
    _70: '#DADADA',
    _80: '#9B9B9B',
  },
};
