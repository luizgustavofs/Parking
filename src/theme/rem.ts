export const REM_BASE = 16;
