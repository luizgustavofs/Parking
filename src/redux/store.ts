import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { combineReducers } from 'redux';
import { persistStore } from 'redux-persist';

import { persistReducers } from './persistReducer';
import authReducer from './reducers/auth';
import preferencesReducer from './reducers/preferences';
import sessionReducer from './reducers/session';

const combinedReducers = combineReducers({
  auth: authReducer,
  preferences: preferencesReducer,
  session: sessionReducer,
});

export type RootReducer = ReturnType<typeof combinedReducers>;

export const store = configureStore({
  reducer: persistReducers(combinedReducers),
});

const persist = persistStore(store);

type RootState = ReturnType<typeof store.getState>;
export const useAppDispatch = () => useDispatch<typeof store.dispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export { persist };
