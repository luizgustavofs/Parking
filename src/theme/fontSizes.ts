import { REM_BASE } from './rem';

export const fontSizes = {
  h1: 38 / REM_BASE,
  h2: 24 / REM_BASE,
  h3: 16 / REM_BASE,
  h4: 15 / REM_BASE,
  h5: 12 / REM_BASE,
};
