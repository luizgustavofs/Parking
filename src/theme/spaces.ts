import { REM_BASE } from './rem';

export const spaces = {
  x1: 4 / REM_BASE,
  x2: 8 / REM_BASE,
  x3: 12 / REM_BASE,
  x4: 16 / REM_BASE,
  x6: 24 / REM_BASE,
  x8: 32 / REM_BASE,
  x10: 40 / REM_BASE,
  x11: 44 / REM_BASE,
  x12: 48 / REM_BASE,
  x13: 52 / REM_BASE,
  x14: 56 / REM_BASE,
  x16: 64 / REM_BASE,
  x18: 72 / REM_BASE,
  x20: 80 / REM_BASE,
  x22: 88 / REM_BASE,
  x24: 96 / REM_BASE,
  x30: 120 / REM_BASE,
  x40: 160 / REM_BASE,
  x50: 200 / REM_BASE,
};
