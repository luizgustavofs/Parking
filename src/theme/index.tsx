import { ReactNode } from 'react';

import { ThemeProvider } from 'styled-components';

import { colors } from './colors';
import { fontSizes } from './fontSizes';
import { spaces } from './spaces';

export const theme = {
  colors,
  spaces,
  fontSizes,
};

const ProjectThemeProvider: React.FC<{ children: ReactNode }> = ({
  children,
  ...props
}) => {
  return (
    <ThemeProvider theme={theme} {...props}>
      {children}
    </ThemeProvider>
  );
};

export { ProjectThemeProvider };
