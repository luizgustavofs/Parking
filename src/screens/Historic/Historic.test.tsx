import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import HistoricScreen from '../../pages/historic';
import { store } from '../../redux/store';
import { replaceMock } from '../../tests/setupTests';
import { theme } from '../../theme';

describe('HistoricScreen', () => {
  it('should render correctly', () => {
    const { toJSON } = create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <HistoricScreen />
        </ThemeProvider>
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should realize useEffect to check has plate', () => {
    const {} = render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <HistoricScreen />
        </ThemeProvider>
      </Provider>,
    );

    useEffect: jest.fn(), expect(replaceMock).toHaveBeenCalled();
  });
});
