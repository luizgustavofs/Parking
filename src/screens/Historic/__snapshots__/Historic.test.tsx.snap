// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricScreen should render correctly 1`] = `
<div
  className="sc-hKMtZM bbAbin"
>
  <div
    className="sc-eCYdqJ cTuguu"
  >
    <div>
      <img
        onClick={[Function]}
        src="/arrow_left.svg"
      />
      <h1 />
    </div>
  </div>
</div>
`;
